### PR TITLE
Revert "Revert "Register Torchvision Ops as Cutom Ops""

### PR DESCRIPTION
**Pull request by fmassa**
Reverts pytorch/vision#1316 , and adds lazy loading for now (which hasn't been added yet but will be in the next commit)

### CODE REVIEW

**Comment by fmassa** [comment]
@peterjc123 do you know the reason for the Windows failure in https://circleci.com/gh/pytorch/vision/6983?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link ?

```
build\temp.win-amd64-3.6\Release\tools\miniconda3\conda-bld\torchvision_1568036619998\work\torchvision\csrc\custom_ops\_custom_ops.cp36-win_amd64.lib : fatal error LNK1120: 1 unresolved externals
```

File: torchvision/csrc/vision.cpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #endif
 
 PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
```
Comment (peterjc123) [inline torchvision/csrc/vision.cpp]: @fmassa Have you specified this macro for MSVC?

**Comment by peterjc123** [comment]
@fmassa  Does this post helps: https://stackoverflow.com/questions/49299905/error-lnk2001-unresolved-external-symbol-pyinit?

File: setup.py [hunk 2]
```diff
@@ -138,7 +147,14 @@ def get_extensions():
             include_dirs=tests_include_dirs,
             define_macros=define_macros,
             extra_compile_args=extra_compile_args,
-        )
+        ),
+        extension(
+            "torchvision._custom_ops",
```
Comment (fmassa) [inline setup.py]: Humm, I'm wondering if this is the reason why the tests are failing on Windows with
```
LINK : error LNK2001: unresolved external symbol PyInit__custom_ops
build\temp.win-amd64-3.6\Release\tools\miniconda3\conda-bld\torchvision_1568036619998\work\torchvision\csrc\custom_ops\_custom_ops.cp36-win_amd64.lib : fatal error LNK1120: 1 unresolved externals
```

**Comment by lara-hdr** [comment]
@fmassa thank you for fixing this!!
this might be an explanation of what is happening : https://stackoverflow.com/a/34777427/11767360 .
I'll try this fix in the link to see if it works on windows.

**Comment by fmassa** [comment]
Subsumed by https://github.com/pytorch/vision/pull/1320

**Pull request closed without merging**